types/node/test.d.ts

```diff
@@ -528,6 +535,12 @@ declare module "node:test" {
          * @param message Message to be reported.
          */
         diagnostic(message: string): void;
+        /**
+         * The absolute path of the test file that created the current test. If a test file imports
+         * additional modules that generate tests, the imported tests will return the path of the root test file.
+         * @since v22.6.0
+         */
+        readonly filePath: string | undefined;
```

Comment:
This property is `undefined` if there is no path associated with the test, _eg._ if running in REPL.